cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -61,7 +61,13 @@ protected TableId getTableId() {
 
   @Nullable
   private QueueName fromRowKey(byte[] oldRowKey) {
-    QueueName queueName = QueueName.from(oldRowKey);
+    QueueName queueName;
+    try {
+      queueName = QueueName.from(oldRowKey);
```

Comment:
This wouldn't solve the case where the upgrade tool crashes (or is cancelled) mid-upgrade, right (half the rows converted, other half not).